great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -1557,3 +1838,49 @@ def _get_evaluation_dependencies(
             metric_dependencies=None,
         )
         return dependencies
+
+
+class MulticolumnMapMetricProvider(MapMetricProvider):
+    condition_domain_keys = (
+        "batch_id",
+        "table",
+        "column_list",
+        "row_condition",
+        "condition_parser",
+        "ignore_row_if",
+    )
+    function_domain_keys = (
+        "batch_id",
+        "table",
+        "column_list",
+        "row_condition",
+        "condition_parser",
+        "ignore_row_if",
+    )
+    condition_value_keys = tuple()
+    function_value_keys = tuple()
+
+    @classmethod
+    def _get_evaluation_dependencies(
+        cls,
+        metric: MetricConfiguration,
+        configuration: Optional[ExpectationConfiguration] = None,
+        execution_engine: Optional[ExecutionEngine] = None,
+        runtime_configuration: Optional[dict] = None,
+    ):
+        dependencies: dict = super()._get_evaluation_dependencies(
+            metric=metric,
+            configuration=configuration,
+            execution_engine=execution_engine,
+            runtime_configuration=runtime_configuration,
+        )
+        table_domain_kwargs: dict = {
+            k: v for k, v in metric.metric_domain_kwargs.items() if k != "column_list"
+        }
```

Comment:
This is an interesting phrase: shouldn't row_condition, for example, also be excluded here?

Reply:
@jcampbell The `row_condition` key should not be excluded, because it is an allowed key of the `TABLE` metric domain.

Reply:
So, is the following an accurate characterization english phrasing of this line, "based on the knowledge of *our* domain_kwargs, the only one we process that is not a valid domain kwarg for the table metric domain type is 'column_list'"?

If so (I think it is), then this seems perfect, we shouldn't change it.

Incidentally...I'm going to look into accessor_keys a bit more; I'm not sure that fits the paradigm well of our tighter definition of domain_kwargs, especially given our conversation yesterday about potentially providing a convenience wrapper around `get_compute_domain`

Reply:
@jcampbell I agree with both points above.  There are extremely important, so let me summarize them below:
1. It is crucial to insure that the domain keys match the metric domain type.  To accomplish this, we use our knowledge of the appropriate domain keys for the given metric domain type of interest.  When we do not handle this properly, we will see warnings from the `ExecutionEngine` subclasses, which tightly validate this relationship.
2. I would also like to brainstorm the `accessor_domain_kwargs` some more.  Here is the rationale: currently, I do not see them used in the codebase.  Would it make sense to consider removing them entirely?  If we did, then we could push all computational optimization down to the `ExecutionEngine` implementations and the accessor logic to the partials (such as, for example, the existing condition partials).  Happy to discuss.

Thank you.